### DISCUSSION TITLE
How to get the stiffness of 15MW wind turbine platform during free decay

### DISCUSSION THREAD

**Discussion by YuYuJou** [OpenFAST/openfast#2433] [Q&A]
I know it is a stupid question
I want to output the stiffness of the platform, but don't know the code and where to put

**Comment by jjonkman** [comment]
Dear @YuYuJou,

What are you referring to when you refer to the "stiffness of the platform"?  Are you referring to mooring stiffness, hydrostatic stiffness, gravitational stiffness associated with body weight, aerodynamic stiffness, elastic stiffness or some combination of the above?  In what mode of motion (surge, heave, pitch, etc.) do you want this stiffness?

Best regards,

**Reply by YuYuJou** [reply]
I want to know the horizontal stiffness.
I have to do that on 6 degrees, and I'm trying to add extra force on it to get the displacement to calculate, but I still don't know which document I have to use?

**Reply by jjonkman** [reply]
Dear @YuYuJou,

I'm still not sure what you mean.  You mention "horizontal stiffness", which suggests to me that you want to know the surge and sway stiffness; perhaps yaw as well.

What do you mean by "6 degrees"?  Are you referring to the 6-DOF motion of the floater, so, effectively, you want a 6x6 stiffness matrix?

Do you have aerodynamics enabled?

Best regards,

**Comment by YuYuJou** [comment]
Yes, I want to know the 6-DOF motion of the floater and I have aerodynamics

**Reply by jjonkman** [reply]
Dear @YuYuJou,

I'm still not sure I understand you, but I gather that you want to obtain the total 6x6 stiffness matrix of the rigid IEA 15-MW RWT atop the Volturnus semisubmersible, including the combination of mooring stiffness, hydrostatic stiffness, gravitational stiffness associated with body weight, and aerodynamic stiffness; is that correct?

You can obtain the total 6x6 stiffness matrix through a linearization analysis in OpenFAST.  With only the 6 platform DOFs in ElastoDyn enabled (and all other structural DOFs disabled), with MAP++ used for the mooring system, with state-space hydrodynamics disabled in HydroDyn, and dynamic wake and unsteady airfoil aerodynamics disabled (but frozen wake enabled) in AeroDyn, a full-system linearization analysis in OpenFAST will produce a 12x12 state matrix A, whose lower-left quadrant will represent -M^-1*K of the full system, where M is the 6x6 mass matrix and K is the 6x6 stiffness matrix of the rigid-body system.  So, to derive K, you must multiple this quadrant by -M.  You can obtain the rigid 6x6 mass matrix M as described in the following OpenFAST issue: https://github.com/OpenFAST/openfast/issues/2422.

Best regards,
